src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClientExtensions.cs

```diff
@@ -92,7 +92,7 @@ public static PSRoleAssignment ToPSRoleAssignment(this RoleAssignment role, Auth
                     ObjectId = adObject.Id
                 };
             }
-            else
+            else if (!excludeAssignmentsForDeletedPrincipals)
```

Comment:
In 'ToPSRoleAssignment' method, 'options' object is not passed, so this will still be like a bool param. Not changing the condition, else it will look not clean.